src/Grpc.AspNetCore.Server/Internal/DefaultServerCallContext.cs

```diff
@@ -0,0 +1,96 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Microsoft.AspNetCore.Http;
+
+namespace Grpc.AspNetCore.Server.Internal
+{
+    internal class DefaultServerCallContext : ServerCallContext
+    {
+        private Lazy<string> _peer;
+
+        internal DefaultServerCallContext(HttpContext httpContext)
+        {
+            HttpContext = httpContext;
+
+            var connection = HttpContext.Connection;
+            if (connection.RemoteIpAddress == null)
+            {
+                _peer = new Lazy<string>();
```

Comment:
Lazy is overkill. Just make it a string that is default to null and add an if check.

Reply:
I was hoping to avoid synthesizing this string unless it's called.